src/backend/codegen/include/codegen/utils/codegen_utils.h

```diff
@@ -940,21 +963,80 @@ llvm::Value* CodeGenerator::GetPointerToMemberImpl(
 }
 
 // ----------------------------------------------------------------------------
-// Implementation of CodeGenerator::RecordNamedExternalFunction()
+// Implementation of CodeGenUtils::RecordNamedExternalFunction()
 
 template <typename ReturnType, typename... ArgumentTypes>
-void CodeGenerator::RecordNamedExternalFunction(const std::string& name) {
+void CodeGenUtils::RecordNamedExternalFunction(const std::string& name) {
   NamedExternalFunction named_external_fn{name,
                                           GetAnnotatedType<ReturnType>(),
                                           {}};
-  code_generator_detail::TypeVectorBuilder<ArgumentTypes...>
+  codegen_utils_detail::TypeVectorBuilder<ArgumentTypes...>
       ::AppendAnnotatedTypes(this,
                              &named_external_fn.argument_types);
 
   named_external_functions_.emplace_back(std::move(named_external_fn));
 }
 
+
+// ----------------------------------------------------------------------------
+// Implementation of CodeGenUtils::GetTypeDefFunctionPointer &
+// CodeGenUtils::CreateFunctionPointer().
+
+namespace codegen_utils_detail {
+/**
+ * @brief Templated helper class that provides a static method which
+ *        unpack the return type, argument types and call the respective
+ *        method
+ *
+ * @tparam FunctionType Function type.
+ * @tparam MethodPtr A pointer-to-method.
+ **/
+template <typename FunctionType>
+class FunctionTypeUnpacker;
+
+/**
+ * @brief Partial specialization of FunctionTypeUnpacker
+ * GetFunctionPointerHelper - Call codegnerator's GetFunctionPointer
+ * CreateFunction
+ **/
+template<typename ReturnType, typename... ArgumentTypes>
+class FunctionTypeUnpacker<ReturnType(*)(ArgumentTypes...)> {
+ public:
+  using R = ReturnType;
```

Comment:
It is used by GetFunctionPointerTypeDef and CreateFunctionTypeDef. around line # 1013